cmd/ctr/events.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	gocontext "context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	containersapi "github.com/containerd/containerd/api/services/containers"
+	eventsapi "github.com/containerd/containerd/api/services/events"
+	executionapi "github.com/containerd/containerd/api/services/execution"
+	"github.com/containerd/containerd/api/types/event"
+	"github.com/urfave/cli"
+)
+
+var eventsCommand = cli.Command{
+	Name:  "events",
+	Usage: "display containerd events",
+	Action: func(context *cli.Context) error {
+		eventsClient, err := getEventsService(context)
+		if err != nil {
+			return err
+		}
+		events, err := eventsClient.EventStream(gocontext.Background(), &eventsapi.EventStreamRequest{})
+		if err != nil {
+			return err
+		}
+		w := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		for {
+			e, err := events.Recv()
+			if err != nil {
+				return err
+			}
+
+			out, err := getEventOutput(e)
+			if err != nil {
+				return err
+			}
+
+			if _, err := fmt.Fprintf(w,
+				"%s\t%s",
+				e.Timestamp,
+				e.Topic,
+			); err != nil {
+				return err
+			}
+			if _, err := fmt.Fprintf(w, "\t%s\n", out); err != nil {
+				return err
+			}
+			if err := w.Flush(); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	},
+}
+
+func getEventOutput(evt *event.Envelope) (string, error) {
+	var r map[string]interface{}
+	// if value is not json; ignore
+	_ = json.Unmarshal(evt.Event.Value, &r)
```

Comment:
the result of that Unmarshalling, if successful, is never used, is something missing?